src/main/scala/zio/interop/reactivestreams/Adapters.scala

```diff
@@ -109,7 +109,9 @@ object Adapters {
             .toManaged(_.shutdown)
       p <- Promise
             .make[Throwable, (Subscription, Queue[Exit[Option[Throwable], A]])]
-            .toManaged(p => ZIO.whenM(p.isDone)(p.await.fold(_ => UIO.unit, { case (sub, _) => UIO(sub.cancel()) })))
```

Comment:
Ouch. The `fold` was throwing away the finalizer?

Reply:
yes. Dropped an `M` 🙄.

Reply:
The downside of `UIO[Any]` for finalizers..